server/src/main/scala/org/http4s/server/middleware/authentication/BasicAuthentication.scala

```diff
@@ -13,18 +13,23 @@ import scalaz.concurrent.Task
  * @param store A partial function mapping (realm, user) to the
  *              appropriate password.
  */
-class BasicAuthentication(realm: String, store: AuthenticationStore) extends Authentication {
+object basicAuth {
```

Comment:
I introduced this change, and I'd like to hear others' feedback.

A middleware is a function, which implies lowercase, and also often an object, which implies uppercase.  We have inconsistent conventions through the library.  I prefer lowercase when it's just an apply, but when it comes with its own types, uppercase starts to feel more right.